clients/wallet/src/horizon.rs

```diff
@@ -374,79 +458,71 @@ pub(crate) struct HorizonFetcher<C: HorizonClient> {
 	client: C,
 	is_public_network: bool,
 	vault_account_public_key: PublicKey,
+	/// keeps track of the latest sequence number
+	last_sequence: SequenceNumber,
```

Comment:
What's the sequence number for?

Reply:
Currently, it has no use. I figured this will come handy in the future.

The problem with cursors (esp. a transaction's paging token) is that it returns a different result depending on: the **_limit_** and the **_order_** and the time of day. 

a [next cursor of 23892903079936 with limit 1 and order asc](https://horizon-testnet.stellar.org/accounts/GCXQXD3EDC7YILNAHRJLYR53MGQU4V5RXB7JHGPVS3HQHLBJDWCJHVIB/transactions?&limit=1&order=asc) actually returns 
```
{
  "type": "https://stellar.org/horizon-errors/bad_request",
  "title": "Bad Request",
  "status": 400,
  "detail": "The request you sent was invalid in some way.",
  "extras": {
    "invalid_field": "cursor",
    "reason": "cursor: invalid value"
  }
}
```
however if you set the order to desc, the result is:
```
{
  "_links": {
    "self": {
      "href": "https://horizon-testnet.stellar.org/accounts/GCXQXD3EDC7YILNAHRJLYR53MGQU4V5RXB7JHGPVS3HQHLBJDWCJHVIB/transactions?cursor=23892903079936&limit=200&order=desc"
    },
    "next": {
      "href": "https://horizon-testnet.stellar.org/accounts/GCXQXD3EDC7YILNAHRJLYR53MGQU4V5RXB7JHGPVS3HQHLBJDWCJHVIB/transactions?cursor=23892903079936&limit=200&order=desc"
    },
    "prev": {
      "href": "https://horizon-testnet.stellar.org/accounts/GCXQXD3EDC7YILNAHRJLYR53MGQU4V5RXB7JHGPVS3HQHLBJDWCJHVIB/transactions?cursor=23892903079936&limit=200&order=asc"
    }
  },
  "_embedded": {
    "records": []
  }
}
```

In other words, saving the cursor alone is not enough. 
However, if we save the last known [sequence number](https://developers.stellar.org/docs/glossary#sequence-number) before a sudden shutdown, we can actually figure out how far away the vault is from the latest transaction. 

We could compare against the `https://horizon-testnet.stellar.org/accounts/<pub_account>`'s _**`sequence`**_ field, and know whether or not the last known transaction was actually executed or not.

Anyway, I'm not yet sure how/when/where to compare it, but i know it's sort of helpful. I hope. 
Say the _`execute_open_request`_: there's no "time" to identify when to stop iterating over a list of transactions. If we have the last known sequence number, we can stop calling `.next()` in the loop by checking whether or not the transaction's sequence number happened too long ago; and that it shouldn't be bothered with (hence breaking the loop entirely because the next transactions are useless. Why fetch 10 hours old transactions, when the vault halted only 9 hours ago?)

Reply:
actually, not sure if it should be in HorizonFetcher or the wallet. 
But i'm not sure whether it's best to share the wallet to `listen_for_new_transactions()`.